### PR TITLE
Remove UserTrackingContext and refactor usages to Portal Context

### LINKED ISSUE #726: Migrate UserContext
Need to migrate the UserContext out of CosmosDB and into SQL. All of the fields tracked by the user preferences should be in the `PortalUser` class from the new telemetry model update.

Will likely need a recent links table

### PR DESCRIPTION
**Pull request by nicksalt**
In this PR, I have completely removed the `UserTrackingContext`. I am aware that there was a previous discussion about potentially utilizing it as a fallback mechanism to prevent users from losing their language preferences. If required, I can reintroduce the `UserTrackingContext`; removing it in this iteration was mainly to facilitate the refactoring of existing usages to rely on the portal context instead.

Closes #726

### CODE REVIEW

File: Portal/src/Datahub.Core/Migrations/Core/DatahubProjectDBContextModelSnapshot.cs [hunk 1]
```diff
@@ -1664,6 +1664,35 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.ToTable("OnboardingApps");
                 });
 
+            modelBuilder.Entity("Datahub.Core.Model.UserTracking.UserRecent", b =>
+                {
+                    b.Property<string>("UserId")
+                        .HasColumnType("nvarchar(450)");
```
Comment (iroelperez) [inline Portal/src/Datahub.Core/Migrations/Core/DatahubProjectDBContextModelSnapshot.cs]: This is a bit too large string field for a user ID, I think if you don't specify the length, they default to 450.

**Review by iroelperez — APPROVED** [review]

**Review by yjmrobert — CHANGES_REQUESTED** [review]
Nice work, a few changes related to what's coming down the pipeline in the next few iterations

- Could we have the FKs link off the `PortalUser` int id
- Since we're switching from NoSQL, we can probably drop `UserRecent` holding class and just have the two DbSets for recent links and settings

It's been awhile since I've dug around there so let me know if I'm off base w/ the requested changes

**Comment by yjmrobert** [comment]
> 

It's been awhile since I've dug around there so let me know if I'm off base w/ the requested changes

File: Portal/src/Datahub.Core/Migrations/Core/DatahubProjectDBContextModelSnapshot.cs (hunk 1 above)
Comment (nicksalt) [reply Portal/src/Datahub.Core/Migrations/Core/DatahubProjectDBContextModelSnapshot.cs]: Resolved in d8c280b

**Review by iroelperez — APPROVED** [review]

**Review by yjmrobert — APPROVED** [review]

**Pull request merged**